### DISCUSSION TITLE
Random error:  Cannot read properties of null (reading 'fingerprint')

### DISCUSSION THREAD

**Discussion by miricristiana** [livewire/livewire#5432] [7. Bugs (archived)]
This error comes up at a completely random timing. We have a setup where we can use a couple of toggle buttons like this:

```html
<div wire:key="{{ $tenant->uuid }}" class="flex flex-col relative w-full mb-5">
    <dd class="mt-1 text-sm text-gray-900">
        @if (method_exists($tenant, 'modules'))
        @foreach ($tenant->modules()->get() as $module)
        <h4 class="my-1.5">
        {{ $module->name }}
        </h4>
            @foreach ($module->roles()->get() as $role)
                <button wire:key="{{ $role->uuid }}" wire:click="toggleRole('{{$role->uuid}}', '{{$tenant->uuid}}')" class="text-xs font-semibold inline-block py-1 px-2 uppercase rounded-full  uppercase last:mr-0 mr-1 outline outline-1 outline-green-600 select-none{{$user->roles()->get()->contains($role) ? ' bg-green-600 text-white ' : ' bg-white  text-green-600'}}"> {{ $role->name }} </button>
            @endforeach
        @endforeach
        @else
        <p>No modules assigned to tenant</p>
        @endif
    </dd>

    <div class="absolute inset-0 z-10 flex flex-col items-center justify-end overflow-y-auto bg-gray-600 bg-opacity-50 sm:justify-start -mx-4 -mb-4 {{$locked? '' : 'hidden'}}">
    </div>
</div>
```

Clicking the button that does "toggleRole" works properly most of the time, but breaks down inevitably and completely randomly.

We searched the internet for solutions, but didn't find anything that helped. All solutions were about changing the `wire:key`. Something like changing `wire:key` to `:wire:key` for example.

This must be a bug. If we can provide more information, please let us know. We will gladly help. This bug has taken hours of our time already.

```
Uncaught (in promise) TypeError: Cannot read properties of null (reading 'fingerprint')
    at new Component (livewire.js?id=de3fca26689cb5a39af4:13:115237)
    at onNodeAdded (livewire.js?id=de3fca26689cb5a39af4:13:124932)
    at callHook (livewire.js?id=de3fca26689cb5a39af4:13:84902)
    at handleNodeAdded (livewire.js?id=de3fca26689cb5a39af4:13:86375)
    at livewire.js?id=de3fca26689cb5a39af4:13:89791
    at morphEl (livewire.js?id=de3fca26689cb5a39af4:13:90339)
    at livewire.js?id=de3fca26689cb5a39af4:13:88518
    at morphEl (livewire.js?id=de3fca26689cb5a39af4:13:90339)
    at livewire.js?id=de3fca26689cb5a39af4:13:91400
    at Component.value (livewire.js?id=de3fca26689cb5a39af4:13:123298)
```

**Comment by dionsnoeijen** [comment]
Is there anything more we should provide? This issue remains, and we cannot find a solution.

**Comment by joshhanley** [comment]
@miricristiana and @dionsnoeijen generally this issue seems to happen if the server responds with a HTML response instead of a JSON response.

You can check this by opening up dev tools to the network tab, refreshing the page, then triggering the request that causes this error. If you then click on that request, and have a look at the response tab. It should be JSON. Can you screenshot what you see?

**Reply by dionsnoeijen** [reply]
Thanks for the suggestion. We are trying to reproduce it, but it's not coming back. I will try more because we didn't change anything. What I can say is that we already checked the response before and it was json. Never saw html returned from the server.

**Comment by rodrigoross** [comment]
I have a component with the same error when I try to fire a request.

Here's the print of the Response tab, and the server is returning a JSON normally.
![image](https://user-images.githubusercontent.com/25902368/219700555-97e82a8b-0432-4fbc-80b5-97ae55311b21.png)
